### ISSUE TITLE
Early initializers can crash Scala 2.13.4

### ISSUE THREAD

**Issue by pweisenburger** [scala/bug#12331]
The following code compiles fine with Scala 2.12.x and 2.13.0, 2.13.1, 2.13.2 and 2.13.3. Compiling it with Scala 2.13.4 throws an `java.lang.UnsupportedOperationException: Position.point on NoPosition` during compilation:

```scala
class A extends scala.annotation.StaticAnnotation

trait T { val x: Int }

@A class C extends { val x = 0 } with T
```

I know early initializers will be replaced by trait parameters. As long as we do not have them in Scala, would be nice if code like this will keep working.

<details>
  <summary>Stack trace</summary>

```
java.lang.UnsupportedOperationException: Position.point on NoPosition
scala.reflect.internal.util.Position.fail(Position.scala:24)
scala.reflect.internal.util.UndefinedPosition.point(Position.scala:102)
scala.reflect.internal.util.UndefinedPosition.point(Position.scala:97)
scala.reflect.internal.Positions$WrappingPosAccumulator.result(Positions.scala:78)
scala.reflect.internal.Positions.wrappingPos(Positions.scala:64)
scala.reflect.internal.Positions.wrappingPos$(Positions.scala:49)
scala.reflect.internal.SymbolTable.wrappingPos(SymbolTable.scala:28)
scala.reflect.internal.Positions.wrappingPos(Positions.scala:48)
scala.reflect.internal.Positions.wrappingPos$(Positions.scala:48)
scala.reflect.internal.SymbolTable.wrappingPos(SymbolTable.scala:28)
scala.reflect.internal.TreeGen.mkTemplate(TreeGen.scala:429)
scala.reflect.internal.ReificationSupport$ReificationSupportImpl$SyntacticClassDef$.apply(ReificationSupport.scala:358)
scala.reflect.internal.MacroAnnotionTreeInfo$PatchedSyntacticClassDef$.apply(TreeInfo.scala:1121)
scala.reflect.internal.MacroAnnotionTreeInfo.$anonfun$getAnnotationZippers$1(TreeInfo.scala:1031)
scala.reflect.internal.MacroAnnotionTreeInfo.loop$3(TreeInfo.scala:1029)
scala.reflect.internal.MacroAnnotionTreeInfo.getAnnotationZippers(TreeInfo.scala:1113)
scala.reflect.internal.MacroAnnotionTreeInfo.getAnnotationZippers$(TreeInfo.scala:1026)
scala.tools.nsc.typechecker.MacroAnnotationNamers$MacroAnnotationNamer.finishSymbol(MacroAnnotationNamers.scala:188)
scala.tools.nsc.typechecker.MacroAnnotationNamers$MacroAnnotationNamer.dispatch$1(MacroAnnotationNamers.scala:41)
scala.tools.nsc.typechecker.MacroAnnotationNamers$MacroAnnotationNamer.standardEnterSym(MacroAnnotationNamers.scala:47)
scala.tools.nsc.typechecker.AnalyzerPlugins.pluginsEnterSym(AnalyzerPlugins.scala:478)
scala.tools.nsc.typechecker.AnalyzerPlugins.pluginsEnterSym$(AnalyzerPlugins.scala:477)
scala.tools.nsc.Global$$anon$5.pluginsEnterSym(Global.scala:484)
scala.tools.nsc.typechecker.Namers$Namer.enterSym(Namers.scala:287)
scala.tools.nsc.typechecker.Namers$Namer.$anonfun$enterSyms$1(Namers.scala:506)
scala.collection.LinearSeqOps.foldLeft(LinearSeq.scala:169)
scala.collection.LinearSeqOps.foldLeft$(LinearSeq.scala:165)
scala.tools.nsc.typechecker.Namers$Namer.enterSyms(Namers.scala:505)
scala.tools.nsc.typechecker.MacroAnnotationNamers$MacroAnnotationNamer.$anonfun$finishSymbolNotExpandee$1(MacroAnnotationNamers.scala:229)
scala.tools.nsc.typechecker.MacroAnnotationNamers$MacroAnnotationNamer.finishSymbolNotExpandee(MacroAnnotationNamers.scala:221)
scala.tools.nsc.typechecker.MacroAnnotationNamers$MacroAnnotationNamer.finishSymbol(MacroAnnotationNamers.scala:189)
scala.tools.nsc.typechecker.MacroAnnotationNamers$MacroAnnotationNamer.dispatch$1(MacroAnnotationNamers.scala:41)
scala.tools.nsc.typechecker.MacroAnnotationNamers$MacroAnnotationNamer.standardEnterSym(MacroAnnotationNamers.scala:47)
scala.tools.nsc.typechecker.AnalyzerPlugins.pluginsEnterSym(AnalyzerPlugins.scala:478)
scala.tools.nsc.typechecker.AnalyzerPlugins.pluginsEnterSym$(AnalyzerPlugins.scala:477)
scala.tools.nsc.Global$$anon$5.pluginsEnterSym(Global.scala:484)
scala.tools.nsc.typechecker.Namers$Namer.enterSym(Namers.scala:287)
scala.tools.nsc.typechecker.Analyzer$namerFactory$$anon$1.apply(Analyzer.scala:51)
scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:454)
scala.tools.nsc.Global$GlobalPhase.run(Global.scala:401)
scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1515)
scala.tools.nsc.Global$Run.compileUnits(Global.scala:1499)
scala.tools.nsc.Global$Run.compileSources(Global.scala:1491)
scala.tools.nsc.Global$Run.compile(Global.scala:1626)
xsbt.CachedCompiler0.run(CompilerInterface.scala:153)
xsbt.CachedCompiler0.run(CompilerInterface.scala:125)
xsbt.CompilerInterface.run(CompilerInterface.scala:39)
jdk.internal.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
java.base/java.lang.reflect.Method.invoke(Method.java:566)
sbt.internal.inc.AnalyzingCompiler.call(AnalyzingCompiler.scala:248)
sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:122)
sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:95)
sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:91)
scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:186)
sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$3(MixedAnalyzingCompiler.scala:82)
sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$3$adapted(MixedAnalyzingCompiler.scala:77)
sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:215)
sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:77)
sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:146)
sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:343)
sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:343)
sbt.internal.inc.Incremental$.doCompile(Incremental.scala:120)
sbt.internal.inc.Incremental$.$anonfun$compile$4(Incremental.scala:100)
sbt.internal.inc.IncrementalCommon.recompileClasses(IncrementalCommon.scala:180)
sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:98)
sbt.internal.inc.Incremental$.$anonfun$compile$3(Incremental.scala:102)
sbt.internal.inc.Incremental$.manageClassfiles(Incremental.scala:155)
sbt.internal.inc.Incremental$.compile(Incremental.scala:92)
sbt.internal.inc.IncrementalCompile$.apply(Compile.scala:75)
sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:348)
sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:301)
sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:168)
sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:248)
sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:74)
sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:1765)
sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:1738)
scala.Function1.$anonfun$compose$1(Function1.scala:49)
sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
sbt.std.Transform$$anon$4.work(Transform.scala:67)
sbt.Execute.$anonfun$submit$2(Execute.scala:281)
sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
sbt.Execute.work(Execute.scala:290)
sbt.Execute.$anonfun$submit$1(Execute.scala:281)
sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
java.base/java.lang.Thread.run(Thread.java:834)
```
</details>

**Comment by som-snytt** [comment]
Under `-Ymacro-annotations`.

**Comment by SethTisue** [comment]
does anybody want to try to rush a fix for this into 2.13.5? I'm leaving it on the 2.13.5 milestone for now, but I'm not labeling it "blocker"

**Comment by som-snytt** [comment]
"The Hurt Blocker" is one of my favorites.

Relatedly https://github.com/scala/bug/issues/8771 and https://github.com/scala/bug/issues/11293#issuecomment-476450594. I expected https://github.com/scala/bug/issues/11060

The PR is PRovisional.
